src/Compiler/Microsoft.CodeAnalysis.Razor/test/DefaultTagHelperDescriptorFactoryTest.cs

```diff
@@ -324,66 +319,66 @@ public static TheoryData IsEnumData
                                 .AsEnum())
                         .Build()
                 },
                 {
-                    typeof(MultiEnumTagHelper),
-                    TagHelperDescriptorBuilder.Create(typeof(MultiEnumTagHelper).FullName, AssemblyName)
-                        .Metadata(GetMetadata<MultiEnumTagHelper>())
+                    "TestNamespace.MultiEnumTagHelper",
+                    TagHelperDescriptorBuilder.Create("TestNamespace.MultiEnumTagHelper", AssemblyName)
+                        .Metadata(GetMetadata("TestNamespace", "MultiEnumTagHelper"))
                         .TagMatchingRuleDescriptor(ruleBuilder => ruleBuilder.RequireTagName("p"))
                         .TagMatchingRuleDescriptor(ruleBuilder => ruleBuilder.RequireTagName("input"))
                         .BoundAttributeDescriptor(builder =>
                             builder
                                 .Name("non-enum-property")
-                                .Metadata(PropertyName(nameof(MultiEnumTagHelper.NonEnumProperty)))
+                                .Metadata(PropertyName("NonEnumProperty"))
                                 .TypeName(typeof(int).FullName))
                         .BoundAttributeDescriptor(builder =>
                             builder
                                 .Name("enum-property")
-                                .Metadata(PropertyName(nameof(MultiEnumTagHelper.EnumProperty)))
-                                .TypeName(typeof(CustomEnum).FullName)
+                                .Metadata(PropertyName("EnumProperty"))
+                                .TypeName("TestNamespace.CustomEnum")
                                 .AsEnum())
                         .Build()
                 },
                 {
-                    typeof(NestedEnumTagHelper),
-                    TagHelperDescriptorBuilder.Create(typeof(NestedEnumTagHelper).FullName, AssemblyName)
-                        .Metadata(GetMetadata<NestedEnumTagHelper>())
+                    "TestNamespace.NestedEnumTagHelper",
+                    TagHelperDescriptorBuilder.Create("TestNamespace.NestedEnumTagHelper", AssemblyName)
+                        .Metadata(GetMetadata("TestNamespace", "NestedEnumTagHelper"))
                         .TagMatchingRuleDescriptor(ruleBuilder => ruleBuilder.RequireTagName("nested-enum"))
                         .BoundAttributeDescriptor(builder =>
                             builder
                                 .Name("nested-enum-property")
-                                .Metadata(PropertyName(nameof(NestedEnumTagHelper.NestedEnumProperty)))
-                                .TypeName($"{typeof(NestedEnumTagHelper).FullName}.{nameof(NestedEnumTagHelper.NestedEnum)}")
+                                .Metadata(PropertyName("NestedEnumProperty"))
+                                .TypeName($"{"TestNamespace.NestedEnumTagHelper"}.{"NestedEnum"}")
```

Comment:
```suggestion
                                .TypeName("TestNamespace.NestedEnumTagHelper.NestedEnum")
```

Reply:
Good catch, thanks. I had some fancy regex to refactor this file, so no wonder this happened.